### DISCUSSION TITLE
spatial query crossing anti-meridian

### DISCUSSION THREAD

**Discussion by ihowat** [icesat2py/icepyx#512] [Q&A] [answered]
Anyone have a good way to deal with spatial queries spanning either side of 180 deg longitude? Does the api except geographic coordinates in 0-360 deg longitude? If yes, just planning on adding some logic to my code to switch to 360 if subset spans 180.

Update: the api does not except lon > 180: "Invalid longitude value (must be between -180 and 180, inclusive)"

so we have to do something messy like break into two polygons either side of 180 and query separately?

**Comment by JessicaS11** [comment, marked as answer]
Hello @ihowat! icepyx does accept geographic coordinates, but they need to be submitted with an argument that lets icepyx know you're intentionally doing so (versus accidentally submitting your coordinates in the wrong order, which makes it hard to detect this case automatically). The history of this limitation is that it's imposed by CMR/NSIDC, so while we have logic to do the 0-360 to -180-180 conversion for users, it doesn't happen automatically.

Simply adding `xdateline=True` to your Query input (e.g. `region = ipx.Query("ATL03", spatial_extent=my_bounding_box, date_range=my_dates, xdateline=True)`) should do the trick. If this DOESN'T work or gives you unexpected output, please do share your inputs so we can debug and fix it (because it means something's not behaving as expected!).

**Comment by ihowat** [comment]
ok, thanks Jessica. Sorry if I missed that in the docs

**Reply by JessicaS11** [reply]
I just checked out the docs and it's on us for not having it clearly outlined. Could you share where you were looking (e.g. the page links) for info on cross-dateline spatial inputs so we can add some better documentation on it where people are looking for that info (instead of just where I happen to know it is)? Thanks!
